pkg/sdn/plugin/controller.go

```diff
@@ -468,24 +481,30 @@ func (plugin *OsdnNode) AddServiceRules(service *kapi.Service, netID uint32) {
 func (plugin *OsdnNode) DeleteServiceRules(service *kapi.Service) {
 	glog.V(5).Infof("DeleteServiceRules for %v", service)
 
+	// Remove blanket rule allowing subsequent IP fragments; since we don't
+	// have the service VNID here we have to ignore errors since this will
+	// fail when removing services in the admin VNID
```

Comment:
Remove the comment (and the extra EndTransaction) since it will no longer apply, but also FYI it wouldn't cause an error anyway; del-flows doesn't consider it an error if there are no matching flows (which has bitten us in the past).

Actually, we decided that all the flows associated with a service will have the same IP and no other services will use that IP, right? So we don't even need a separate fragment rule, or a loop on ServicePorts; just `DelFlows("table=60, ip, nw_dst=IP")` and that will get everything. (With maybe some further refactoring of the get*ServiceRule functions. Note that it's ok to redundantly specify both `ip` and `tcp` in a rule so you could use that `table,ip,nw_dst` rule as the base of the add rule too.)